rfcs/0031-bazel.md

```diff
@@ -0,0 +1,358 @@
+  its toolchain model, so these must be converted. I've done that with
+  the crosstools-NG toolchain and managed a minimal build of some of
+  the libraries with it.  It will require a little more work and
+  testing, but basically we're in good shape as far as C/C++ cross
+  compilation is concerned.
+* Haven't really looked into Bazel-Ocaml yet.
+
+### Effort Estimates and Timeline
+
+Effort = estimated remaining work effort in hours.
+
+C/C++
+
+| | Platform | Task | Effort | Due | Notes |
+| -- | ----------- | ----------- | ------ | ---------- | ---------------|
+| 1  | all | C/C++ external libs | 1 | Fri 5/15 | See note 1 |
+| 2 | all | libsnark | 3 | Fri 5/15 |
+| 3 | Mac to Linux x86 | cross-platform build | 4 | Fri 5/15 | Code adapted from previous project needs polish and testing |
+| 4 | Mac to Linux arm | cross-platform build | 2 | W 5/20 | Once an x86 toolchain is working adding support for other target archs should be easy |
+| 5 | Mac to Android | cross-platform build | 8 | W 5/20 |Reusing and adapting code from previous project |
+| 6 | linux | configure dev. env | 2 | Fri 5/22 |assuming a docker container can be used as a linux dev environment on Mac |
+| 7 | linux | C/C++ min. bld | 2 | Fri 5/22 |the Mac stuff should work with few changes |
+| 8 | windows | configure dev. env | 4 | TBD | setup VM image, install windows, install tools, etc. |
+| 9 | windows | C/C++ min. bld | 8 | TBD |paths, compiler flags, etc. will take some work, depending on toolchain |
+
+Notes:
+
+1. External libs:
+  * Assumption: what works on the mac will require little or no effort to work on Linux and Windows
+  * libpq (Postgresql) won't build (possible bug in Bazel); workaround is to just install it manually
+
+**OCaml**
+
+| | Platform | Task | Effort | Due | Notes |
+| -- | ----------- | ----------- | ------ | ---------- | ---------------|
+| 10 | all | OCaml/Dune learning curve | 4 | ongoing |  I can usually fake it, since all languages are alike, but darnit, sometimes I have to sit down and actually learn something. |
+| 11 | all | Analyze existing Bazel/OCaml libs | 4 | ongoing | |
+| 12 | all | Develop rules_ocaml | 12 |Sun 5/31 |
+| 13 | all | Implement Bazel support for Coda OCaml code | 16 | Sun 5/31 |
+
+**Golang, CUDA**
+
+| | Platform | Task | Effort | Due | Notes |
+| -- | ----------- | ----------- | ------ | ---------- | ---------------|
+| 14 | all | go-libp2p | 4 | W 5/27  |
+| 15 | all | CUDA code in Coda | 1 | TBD | cuda-fixnum, cuda-prover, etc. |
+
+Note: I don't have a GPU, and I'm not sure if I can compile CUDA code
+without one.  If I can, task 15 will be completed by end of month.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+The learning curve for Bazel is little steep in places, plus it has
+its own languge, Starlark - yet another language to learn.  It's a
+Python dialect, but still, somebody on staff will probably have to be
+tasked with learning the stuff.  On the other hand, once the build
```

Comment:
I wrote a lot of Skylark a few years ago + I am on the core team 👍 . No worries.